src/server.js

```diff
@@ -375,7 +389,7 @@ module.exports = function (classes) {
                 // Try to notify client about failure to authenticate
                 if (Endpoint.hasId(decoded)) {
                   conn.sendReply('Error: Unauthorized', null, decoded.id);
-                  reject(new Error.InvalidParams(UNAUTHORIZED))
+                  return reject(new Error.InvalidParams(UNAUTHORIZED));
```

Comment:
well, it might be an over optimization, but everytime you return something, even if you aren't going to use it, like a long lived object such as a promise, you might end up with weird GC cases and hard to pinpoint memory leaks. like returning the `arguments` from a function, it's a long lived object that holds external objects now inside an internal state. 

their point here in returning the callback is just for the sake of being consistent I guess. if the callback (or reject/resolve) is the last statement on the function body, it makes no sense to return